virtcontainers/pkg/oci/utils.go

```diff
@@ -406,25 +404,23 @@ func (spec *CompatOCISpec) SandboxID() (string, error) {
 	return "", fmt.Errorf("Could not find sandbox ID")
 }
 
-func vmConfig(ocispec CompatOCISpec, config RuntimeConfig) (vc.Resources, error) {
-	resources := config.VMConfig
-
+func updateVMConfig(ocispec CompatOCISpec, config *RuntimeConfig) error {
 	if ocispec.Linux == nil || ocispec.Linux.Resources == nil {
-		return resources, nil
+		return nil
 	}
 
 	if ocispec.Linux.Resources.Memory != nil &&
 		ocispec.Linux.Resources.Memory.Limit != nil {
 		memBytes := *ocispec.Linux.Resources.Memory.Limit
 		if memBytes <= 0 {
-			return vc.Resources{}, fmt.Errorf("Invalid OCI memory limit %d", memBytes)
+			return fmt.Errorf("Invalid OCI memory limit %d", memBytes)
 		}
 		// Use some math magic to round up to the nearest Mb.
 		// This has the side effect that we can never have <1Mb assigned.
-		resources.Memory = uint((memBytes + (1024*1024 - 1)) / (1024 * 1024))
+		config.HypervisorConfig.DefaultMemSz = uint32((memBytes + (1024*1024 - 1)) / (1024 * 1024))
```

Comment:
@bergwolf I think we should rename this to not include "Default" anymore, considering how this var is being used now.